checker/build.gradle

```diff
@@ -434,6 +438,56 @@ task ainferTestCheckerGenerateAjava(type: Test) {
 
     doLast {
         copyNonannotatedToAnnotatedDirectory("ainfer-testchecker")
+
+        // A crash occurs because of a bug in Java Parser when trying to read the .ajava
+        // file produced for these tests. Reinstate them when https://github.com/javaparser/javaparser/pull/3527
+        // has been merged and released, and we've updated to a JavaParser version with that fix.
+        delete("tests/ainfer-testchecker/annotated/all-systems/java8/memberref/Issue946.java")
+        delete("tests/ainfer-testchecker/annotated/all-systems/java8/memberref/Receivers.java")
+
+        // This test must be deleted, because otherwise an error about a missing type in an
+        // ajava file is issued. The test itself shouldn't be run as an all-systems test; see
+        // the copy in the non-annotated WPI tests for an explanation.
+        delete("tests/ainfer-testchecker/annotated/all-systems/java8/memberref/Purity.java")
```

Comment:
I agree that this is unfortunate. Your suggestion of moving it into a separate test suite might work, but I think it would add an equally-annoying number of complications. In particular, if we want to keep the current behavior (i.e., having all checkers that run on the all-systems tests run on this test, too), we'd need to enable the new test suite for each of those checkers individually (and we'd need to remember to add it to all checkers we create in the future, too, which I think is even worse). Mechanically, I think that would require adding something like `"purity"` to the list of test suites in all test runners that currently include `"all-systems"`.

Personally, I prefer what's in this PR now (only WPI is a special snowflake, behavior for "standard" checkers is unchanged) to the alternative (WPI is a slightly-less special snowflake, we need to make a small change in the test setup for a large number of other checkers).

Reply:
That said, if you strongly prefer the alternative I proposed above (or can think of a better plan!), I'm willing to do it. I just think it will be more annoying to maintain, long term.

Reply:
I went with Suzanne's first suggestion of moving the `doNotTypecheck` call into `AinferGeneratePerDirectoryTest`, which definitely reduces duplication.